processor/schemaprocessor/internal/schema/schema.go

```diff
@@ -0,0 +1,30 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+// Schema abstracts the parsed translation file
+// into a useable
+type Schema interface {
+	SupportedVersion(version *Version) bool
```

Comment:
Is this a blocking call? The RequestSchema() comments say that "Operations on the returned schema will be blocked".

If this is a blocking call then we need to pass a context to be able to cancel it.

Reply:
Is this the only method on this interface? How are the actual translation rules accessed?

Reply:
At the time of submitting, I hadn't fully worked out the interface so I left processing part for a follow up PR.

However, I have done a majority of the work now, Just working on the revert pathway now and trying to make it as simply as possible.